RealmSwift/Impl/Persistable.swift

```diff
@@ -44,7 +44,7 @@ extension NSDate: _HasPersistedType {
 }
 
 // A type which can be stored by the @Persisted property wrapper
-public protocol _Persistable: _RealmSchemaDiscoverable, _HasPersistedType where PersistedType: _Persistable {
+public protocol _Persistable: _RealmSchemaDiscoverable, _HasPersistedType where PersistedType: _Persistable, PersistedType.PersistedType.PersistedType == PersistedType.PersistedType {
```

Comment:
Why is it three levels deep on the lhs? Will this cover all cases?

Reply:
If `Foo` maps to `Bar`, which is an `EmbeddedObject` subclass, `Foo.PersistedType` is `Bar`, `Foo.PersistedType.PersistedType` is `EmbeddedObject`, and `Foo.PersistedType.PersistedType.PersistedType` is `EmbeddedObject`. The simpler `PersistedType.PersistedType == PersistedType` doesn't work because `EmbeddedObject` subclasses can't do `PersistedType = Self`.